event/event.go

```diff
@@ -124,16 +186,32 @@ func (b *bus) Publish(ctx context.Context, topic Type, data interface{}) error {
 }
 
 // Subscribe requests events from the given topic, returning a channel of those events
-func (b *bus) Subscribe(handler Handler, eventTypes ...Type) {
+func (b *bus) SubscribeTopics(handler Handler, eventTopics ...Topic) {
 	b.lk.Lock()
 	defer b.lk.Unlock()
-	log.Debugf("Subscribe: %v", eventTypes)
+	log.Debugf("Subscribe to topics: %v", eventTopics)
 
-	for _, topic := range eventTypes {
+	for _, topic := range eventTopics {
 		b.subs[topic] = append(b.subs[topic], handler)
 	}
 }
 
+// SubscribeID requests events that match the given sessionID
+func (b *bus) SubscribeID(handler Handler, sessionID string) {
+	b.lk.Lock()
+	defer b.lk.Unlock()
+	log.Debugf("Subscribe to ID: %v", sessionID)
+	b.idSubs[sessionID] = append(b.idSubs[sessionID], handler)
+}
+
+// SubscribeAll requets all events from the bus
+func (b *bus) SubscribeAll(handler Handler) {
+	b.lk.Lock()
+	defer b.lk.Unlock()
+	log.Debugf("Subscribe All")
+	b.allSubs = append(b.allSubs, handler)
+}
+
 // NumSubscribers returns the number of subscribers to the bus's events
 func (b *bus) NumSubscribers() int {
 	b.lk.Lock()
```

Comment:
I know it's not your code and this is used pretty infrequently but this should be an `RLock/RUnlock`